tests/integration/ha_tests/test_ha.py

```diff
@@ -65,6 +83,128 @@ async def test_build_and_deploy(ops_test: OpsTest) -> None:
     await relate_mongodb_and_application(ops_test, mongodb_application_name, application_name)
 
 
+@pytest.mark.abort_on_fail
+async def test_add_units(ops_test: OpsTest, continuous_writes) -> None:
+    """Tests juju add-unit functionality.
+
+    Verifies that when a new unit is added to the MongoDB application that it is added to the
+    MongoDB replica set configuration.
+    """
+    # add units and wait for idle
+    app = await get_application_name(ops_test, APP_NAME)
+    initial_units = [unit.name for unit in ops_test.model.applications[app].units]
+    await scale_application(ops_test, app, len(ops_test.model.applications[app].units) + 2)
+
+    # grab unit hosts
+    hostnames = await get_units_hostnames(ops_test)
+
+    # connect to replica set uri and get replica set members
+    member_hosts = await fetch_replica_set_members(ops_test)
+
+    # verify that the replica set members have the correct units
+    assert set(member_hosts) == set(hostnames), "all members not running under the same replset"
+
+    # verify that the no writes were skipped
+    total_expected_writes = await get_total_writes(ops_test)
+    with await get_mongo_client(ops_test, excluded=initial_units) as client:
+        actual_writes = client[TEST_DB][TEST_COLLECTION].count_documents({})
+    assert total_expected_writes == actual_writes, "writes to the db were missed."
+
+
+@pytest.mark.abort_on_fail
+async def test_scale_down_capablities(ops_test: OpsTest, continuous_writes) -> None:
+    """Tests clusters behavior when scaling down a minority and removing a primary replica."""
+    app = await get_application_name(ops_test, APP_NAME)
+    minority_count = int(len(ops_test.model.applications[app].units) / 2)
+    expected_units = len(ops_test.model.applications[app].units) - minority_count
+
+    # find leader unit
+    leader_unit = await find_unit(ops_test, leader=True)
+
+    # verify that we have a leader
+    assert leader_unit is not None, "No unit is leader"
+
+    # Force delete the leader and scale down
+    await kubectl_delete(ops_test, leader_unit, False)
```

Comment:
The VM test destroys the leader, but in K8S we can't control which unit to scale down. I can delete it manually, to mimic the behaviour, but the StatefulSet would most probably just revive it. Do we need this?

Reply:
Deleting the pod doesn't seem to trigger a Juju election.

Reply:
Pod will be recreated and resume being a leader.